opnode/rollup/driver/step.go

```diff
@@ -120,3 +176,40 @@ func AddBlock(ctx context.Context, logger log.Logger, rpc DriverAPI,
 	logger.Info("updated fork-choice with block")
 	return payload.ID(), nil
 }
+
+// AddBlockForBSS extends the L2 chain by deriving the full execution payload from inputs,
+// and then executing and persisting it.
+//
+// After the step completes it returns the block ID of the last processed L2 block, even if an error occurs.
+func AddBlockForBSS(ctx context.Context, logger log.Logger, rpc DriverAPI, epoch uint64,
```

Comment:
The cleanup approach (other PR) was to have one `AddBlock` that returns the full payload, and then either compute the block ID (as verifier) or the batch (as sequencer) from that right? Or are there other differences?

Reply:
There are other differences, but I can bring back the single add block function. If all you get is the block ID it's impossible to create the batch, but if you have the payload, it's not that hard.